src/v3.rs

```diff
@@ -538,13 +555,33 @@ impl Attachment {
     }
 }
 
+impl ASM {
+    /// Construct a new attachment for this message.
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// The unsubscribe group to associate with this email.
+    pub fn set_group_id(mut self, group_id: u32) -> Self {
+        self.group_id = group_id;
+        self
+    }
+
+    /// A set containing the unsubscribe groups that you would like to be displayed on the unsubscribe preferences page.
```

Comment:
Let's add a check here to see if there are greater than 25 items. This appears to be the max per the documentation.

Reply:
Do you want it to truncate to 25 items or should it return a `Result` so we can handle the error case?

Reply:
I went with using a `SendgridResult` and added a new entry to the `SendgridError` enum.